src/Raven.Server/Rachis/Follower.cs

```diff
@@ -110,7 +110,14 @@ private void FollowerSteadyState()
                     {
                         for (int i = 0; i < appendEntries.EntriesCount; i++)
                         {
-                            entries.Add(_connection.ReadRachisEntry(context));
+                            using (var cts = new CancellationTokenSource())
+                            {
+                                var task = Concurrent_SendAppendEntriesPendingToLeaderAsync(cts, _term, appendEntries.PrevLogIndex);
```

Comment:
Can we do this for the entire batch of commands, rather than for each single one?

Reply:
yes